### PR TITLE
Added support for multi-client

### PR DESCRIPTION
**Pull request by dim**
I've been maintaining https://github.com/bsm/multiredis for a while as it simplifies many use cases and testing scenarios. It has been really useful and I would like to include it into the main repo.

### CODE REVIEW

File: multi.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
+package redis
+
+import "time"
+
+// MultiOptions information is required by MultiClient to establish
+// connections. There are To connect to a cluster
```
Comment (vmihailenco) [inline multi.go]: `There are To connect` seems to be truncated.

**Comment by vmihailenco** [comment]
I think name `Multi` is misleading and confusing, because only one client is created. And comments/docs don't say multi or multiple too which is a bad sign. May be we can find a better name? OpaqueClient? SomeClient? :)

Overall this looks rather specific, but since there are docs and comments I don't mind merging it into master.

**Comment by dim** [comment]
How about `UniversalClient`?

**Comment by vmihailenco** [comment]
Sounds good - please rebase this against master to be merged into master branch.

**Comment by dim** [comment]
Updated, tests on Travis now pass too

**Pull request merged**